### PR TITLE
[ci-app] Add User Provided Git Metadata

### PR DESCRIPTION
**Pull request by juan-fernandez**
## Motivation
Provide a way for users to provide `git` metadata when other methods fail. 

## How
* Read certain environment variables and set them as span tags if they're present. These take precedence over other methods to get `git` metadata. 

**Note**: I don't understand what's happening with `sorbet`'s job. If I run the typecheck in master the same errors show, so I'm not sure if they're related to my changes. Also, the files with errors seem not to be the ones I changed?

### CODE REVIEW

File: lib/datadog/ci/ext/environment.rb [hunk 1]
```diff
@@ -324,6 +327,22 @@ def extract_bitrise(env)
...
+            Datadog::Ext::Git::TAG_BRANCH => env['DD_GIT_BRANCH'],
+            Datadog::Ext::Git::TAG_TAG => env['DD_GIT_TAG'],
+            Datadog::Ext::Git::TAG_COMMIT_MESSAGE => env['DD_GIT_COMMIT_MESSAGE'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_NAME => env['DD_GIT_COMMIT_AUTHOR_NAME'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_EMAIL => env['DD_GIT_COMMIT_AUTHOR_EMAIL'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_DATE => env['DD_GIT_COMMIT_AUTHOR_DATE'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_NAME => env['DD_GIT_COMMIT_COMMITTER_NAME'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_EMAIL => env['DD_GIT_COMMIT_COMMITTER_EMAIL'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_DATE => env['DD_GIT_COMMIT_COMMITTER_DATE']
+          }.reject { |_, v| v.nil? || v.empty? }
```
Comment (delner) [inline lib/datadog/ci/ext/environment.rb:343]: I would recommend using `strip` on the values first, as sometimes users will introduce whitespace characters that will make the string "not empty", e.g.:

```
2.5.1 :002 > ' '.empty?
 => false
 ```

File: lib/datadog/ci/ext/environment.rb [hunk 2]
```diff
@@ -324,6 +327,22 @@ def extract_bitrise(env)
           }
         end
 
+        def extract_user_defined_git(env)
+          {
+            Datadog::Ext::Git::TAG_REPOSITORY_URL => env['DD_GIT_REPOSITORY_URL'],
```
Comment (delner) [inline lib/datadog/ci/ext/environment.rb:332]: All these strings should be frozen, migrated to `Datadog::Ext::Git` and prefixed with `ENV_`

e.g. `'DD_GIT_REPOSITORY_URL'` => `Datadog::Ext::Git::ENV_REPOSITORY_URL = 'DD_GIT_REPOSITORY_URL'.freeze`

File: spec/datadog/ci/ext/environment_spec.rb [hunk 3]
```diff
@@ -118,6 +118,44 @@
...
+            'DD_GIT_COMMIT_COMMITTER_EMAIL' => 'user-committer@provided.com',
+            'DD_GIT_COMMIT_COMMITTER_DATE' => '2021-06-19T18:35:10+00:00',
+          }
+        end
+
+        it 'returns user provided metadata' do
+          is_expected.to eq(
+            {
+              'ci.workspace_path' => "#{Dir.pwd}/spec/datadog/ci/ext/fixtures/git",
+              'git.branch' => 'my-branch',
```
Comment (delner) [inline spec/datadog/ci/ext/environment_spec.rb:144]: Minor suggestion, but you could derive these values from the `env` above e.g. `'git.branch' => env['DD_GIT_BRANCH']`

**Review by delner — CHANGES_REQUESTED** [review]
Overall seems fine, but I think we should migrate some constants to follow convention.

**Review by marcotc — APPROVED** [review]

**Pull request merged**